File: examples/csharp/basic.cs
/*
 * Planet Positions API - Basic Usage Example
 *
 * This example demonstrates the basic usage of the Planet Positions API.
 * API Documentation: https://docs.apiverve.com/ref/planetpositions
 */

using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using System.Linq;

namespace APIVerve.Examples
{
    class PlanetPositionsExample
    {
        private static readonly string API_KEY = Environment.GetEnvironmentVariable("APIVERVE_API_KEY") ?? "YOUR_API_KEY_HERE";
        private static readonly string API_URL = "https://api.apiverve.com/v1/planetpositions";

        /// <summary>
        /// Make a POST request to the Planet Positions API
        /// </summary>
        static async Task<JsonDocument> CallPlanetPositionsAPI()
        {
            try
            {
                using var client = new HttpClient();
                client.DefaultRequestHeaders.Add("x-api-key", API_KEY);

                // Request body
                var requestBody &#x3D; new { planet &#x3D; &quot;Moon&quot;, date &#x3D; &quot;2025-04-15 10:37:00&quot;, lat &#x3D; 37.7749, lon &#x3D; -122.4194, alt &#x3D; 52 };

                var jsonContent = JsonSerializer.Serialize(requestBody);
                var content = new StringContent(jsonContent, Encoding.UTF8, "application/json");

                var response = await client.PostAsync(API_URL, content);

                // Check if response is successful
                response.EnsureSuccessStatusCode();

                var responseBody = await response.Content.ReadAsStringAsync();
                var data = JsonDocument.Parse(responseBody);

                // Check API response status
                if (data.RootElement.GetProperty("status").GetString() == "ok")
                {
                    Console.WriteLine("✓ Success!");
                    Console.WriteLine("Response data: " + data.RootElement.GetProperty("data"));
                    return data;
                }
                else
                {
                    var error = data.RootElement.TryGetProperty("error", out var errorProp)
                        ? errorProp.GetString()
                        : "Unknown error";
                    Console.WriteLine($"✗ API Error: {error}");
                    return null;
                }
            }
            catch (HttpRequestException e)
            {
                Console.WriteLine($"✗ Request failed: {e.Message}");
                return null;
            }
        }

        static async Task Main(string[] args)
        {
            Console.WriteLine("📤 Calling Planet Positions API...\n");

            var result = await CallPlanetPositionsAPI();

            if (result != null)
            {
                Console.WriteLine("\n📊 Final Result:");
                Console.WriteLine(result.RootElement.GetProperty("data"));
            }
            else
            {
                Console.WriteLine("\n✗ API call failed");
            }
        }
    }
}

File: examples/javascript/basic.js
/**
 * Planet Positions API - Basic Usage Example
 *
 * This example demonstrates the basic usage of the Planet Positions API.
 * API Documentation: https://docs.apiverve.com/ref/planetpositions
 */

const API_KEY = process.env.APIVERVE_API_KEY || 'YOUR_API_KEY_HERE';
const API_URL = 'https://api.apiverve.com/v1/planetpositions';

/**
 * Make a POST request to the Planet Positions API
 */
async function callPlanetPositionsAPI() {
  try {
    // Request body
    const requestBody &#x3D; {
    &quot;planet&quot;: &quot;Moon&quot;,
    &quot;date&quot;: &quot;2025-04-15 10:37:00&quot;,
    &quot;lat&quot;: 37.7749,
    &quot;lon&quot;: -122.4194,
    &quot;alt&quot;: 52
};

    const response = await fetch(API_URL, {
      method: 'POST',
      headers: {
        'x-api-key': API_KEY,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(requestBody)
    });

    // Check if response is successful
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const data = await response.json();

    // Check API response status
    if (data.status === 'ok') {
      console.log('✓ Success!');
      console.log('Response data:', data.data);
      return data.data;
    } else {
      console.error('✗ API Error:', data.error || 'Unknown error');
      return null;
    }

  } catch (error) {
    console.error('✗ Request failed:', error.message);
    return null;
  }
}

// Run the example
callPlanetPositionsAPI()
  .then(result => {
    if (result) {
      console.log('\n📊 Final Result:');
      console.log(JSON.stringify(result, null, 2));
    }
  });
